src/open_inwoner/cms/cases/views/status.py

```diff
@@ -290,74 +388,112 @@ def handle_document_upload(self, request, form):
+        else:
+            return self.form_invalid(form)
 
+    def get_success_url(self):
+        return reverse(
+            "cases:case_detail_contact_form", kwargs={"object_id": str(self.case.uuid)}
+        )
 
-class CaseDocumentDownloadView(LogMixin, CaseAccessMixin, View):
-    def get(self, request, *args, **kwargs):
-        if not self.case:
-            raise Http404
+    def register_by_api(self, form):
+        config = OpenKlantConfig.get_solo()
+        assert config.has_api_configuration()
 
-        info_object_uuid = kwargs["info_id"]
-        info_object = fetch_single_information_object_uuid(info_object_uuid)
-        if not info_object:
-            raise Http404
+        klant = fetch_klant_for_bsn(self.request.user.bsn)
+        if klant:
+            self.log_system_action(
+                "retrieved klant for BSN-user", user=self.request.user
+            )
+        else:
+            self.log_system_action(
+                "could not retrieve klant for BSN-user", user=self.request.user
+            )
+            data = {
+                "bronorganisatie": config.register_bronorganisatie_rsin,
+                "voornaam": self.request.user.first_name,
+                "voorvoegselAchternaam": self.request.user.infix,
+                "achternaam": self.request.user.last_name,
+                "emailadres": self.request.user.email,
+                "telefoonnummer": self.request.user.phonenumber,
+            }
+            # registering klanten won't work in e-Suite as it always pulls from BRP (but try anyway and fallback to appending details to tekst if fails)
+            klant = create_klant(data)
+            if klant:
+                self.log_system_action(
+                    "created klant for basic authenticated user",
+                    user=self.request.user,
+                )
+            else:
+                self.log_system_action(
+                    "could not create klant for BSN-user", user=self.request.user
+                )
 
-        # check if this info_object belongs to this case
-        if not fetch_case_information_objects_for_case_and_info(
-            self.case.url, info_object.url
-        ):
-            raise PermissionDenied()
+        # create contact moment
+        contact_moment = form.cleaned_data["contact_moment"]
+        data = {
+            "bronorganisatie": config.register_bronorganisatie_rsin,
+            "tekst": contact_moment,
+            "type": config.register_type,
+            "kanaal": "Internet",
+            "medewerkerIdentificatie": {
+                "identificatie": config.register_employee_id,
+            },
+        }
 
-        # check if this info_object should be visible
-        config = OpenZaakConfig.get_solo()
-        if not is_info_object_visible(info_object, config.document_max_confidentiality):
-            raise PermissionDenied()
+        contactmoment = create_contactmoment(data, klant=klant)
```

Comment:
I still wonder about the situation when we don't get a klant and cannot create one either, in which case we'd loose the connection between the person and the contactmoment. We probably discussed this?